Source/WebCore/layout/formattingContexts/inline/InlineItemsBuilder.cpp

```diff
@@ -878,88 +878,86 @@ void InlineItemsBuilder::handleTextContent(const InlineTextBox& inlineTextBox, I
-            if (!isSegmentBreakCandidate || !shouldPreserveNewline)
-                return false;
-            inlineItemList.append(InlineSoftLineBreakItem::createSoftLineBreakItem(inlineTextBox, currentPosition++));
-            return true;
-        };
-        auto handleWhitespace = [&] {
-            auto stopAtWordSeparatorBoundary = shouldPreserveSpacesAndTabs && style.fontCascade().wordSpacing();
-            auto whitespaceContent = text.is8Bit() ?
-                moveToNextNonWhitespacePosition(text.span8(), currentPosition, shouldPreserveNewline, shouldPreserveSpacesAndTabs, stopAtWordSeparatorBoundary) :
-                moveToNextNonWhitespacePosition(text.span16(), currentPosition, shouldPreserveNewline, shouldPreserveSpacesAndTabs, stopAtWordSeparatorBoundary);
-            if (!whitespaceContent)
-                return false;
-
-            ASSERT(whitespaceContent->length);
-            if (style.whiteSpaceCollapse() == WhiteSpaceCollapse::BreakSpaces) {
-                // https://www.w3.org/TR/css-text-3/#white-space-phase-1
-                // For break-spaces, a soft wrap opportunity exists after every space and every tab.
-                // FIXME: if this turns out to be a perf hit with too many individual whitespace inline items, we should transition this logic to line breaking.
-                inlineItemList.appendUsingFunctor(whitespaceContent->length, [&](size_t offset) {
-                    return InlineTextItem::createWhitespaceItem(inlineTextBox, currentPosition + offset, 1, UBIDI_DEFAULT_LTR, whitespaceContent->isWordSeparator, { });
-                });
-            } else
-                inlineItemList.append(InlineTextItem::createWhitespaceItem(inlineTextBox, currentPosition, whitespaceContent->length, UBIDI_DEFAULT_LTR, whitespaceContent->isWordSeparator, { }));
-            currentPosition += whitespaceContent->length;
-            return true;
+    auto handleSegmentBreak = [&] {
+        // Segment breaks with preserve new line style (white-space: pre, pre-wrap, break-spaces and pre-line) compute to forced line break.
+        auto isSegmentBreakCandidate = text[currentPosition] == newlineCharacter;
+        if (!isSegmentBreakCandidate || !shouldPreserveNewline)
+            return false;
+        inlineItemList.append(InlineSoftLineBreakItem::createSoftLineBreakItem(inlineTextBox, currentPosition++));
+        return true;
+    };
+    auto handleWhitespace = [&] {
+        auto stopAtWordSeparatorBoundary = shouldPreserveSpacesAndTabs && style.fontCascade().wordSpacing();
+        auto whitespaceContent = text.is8Bit() ?
+            moveToNextNonWhitespacePosition(text.span8(), currentPosition, shouldPreserveNewline, shouldPreserveSpacesAndTabs, stopAtWordSeparatorBoundary) :
+            moveToNextNonWhitespacePosition(text.span16(), currentPosition, shouldPreserveNewline, shouldPreserveSpacesAndTabs, stopAtWordSeparatorBoundary);
```

Comment:
```suggestion
        auto whitespaceContent = text.is8Bit()
            ? moveToNextNonWhitespacePosition(text.span8(), currentPosition, shouldPreserveNewline, shouldPreserveSpacesAndTabs, stopAtWordSeparatorBoundary)
            : moveToNextNonWhitespacePosition(text.span16(), currentPosition, shouldPreserveNewline, shouldPreserveSpacesAndTabs, stopAtWordSeparatorBoundary);
```

I think this (or something similar) would address the first checker guide error